src/ArrayHelper.php

```diff
@@ -212,6 +212,10 @@ public static function getValue(
             return $key($array, $default);
         }
 
+        if (is_array($array) && (is_int($key) || is_string($key))) {
```

Comment:
```suggestion
        if (is_array($array) && is_scalar($key)) {
```

Have you benched this case?

Reply:
Note: need to import the function or set the global namespace directly
`\is_scalar`